backend/TraderX/src/main/java/cmpe451/group6/rest/transaction/repository/TransactionRepository.java

```diff
@@ -0,0 +1,31 @@
+package cmpe451.group6.rest.transaction.repository;
+
+
+import cmpe451.group6.rest.transaction.model.Transaction;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Date;
+import java.util.List;
+
+public interface TransactionRepository extends JpaRepository<Transaction,Integer> {
+
+    List<Transaction> findByUser_username(String username);
+
+    List<Transaction> findByEquipment_code(String code);
+
+    List<Transaction> findAll();
+
+    int countByUser_username(String username);
+
+    int countByEquipment_code(String code);
+
+    @Query("SELECT COUNT(t) FROM Transaction t ")
+    int countAll();
+
+    @Query("SELECT t FROM Transaction t WHERE t.createdAt BETWEEN  ?1 AND ?2")
+    List<Transaction> findByDateBetween(Date start, Date end);
+
```

Comment:
What is the aim of the function? It seems to return all transactions in the system(made by all users). I think we should implement a function that returns the transactions of a specified user (in a specified time period by start and end)

Reply:
See reply: https://github.com/bounswe/bounswe2019group6/pull/178#discussion_r346695925